### PR TITLE
Auto Json

### PR DESCRIPTION
**Pull request by csquared**
automatically detect and parse json, adding any top-level
keys to the form_hash exposed by rack

### CODE REVIEW

**Comment by Hudon** [comment]
" request body that begins with a `{` to detect and parse json."

json can also start with a "["

**Comment by csquared** [comment]
@Hudon good call -- i added that and tightened up the tests.  also used that `Hash[]` thing to make the array work with the normal params hash :smile:

File: lib/jersey/middleware/auto_json.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    def call(env)
+      req = Rack::Request.new(env)
+      type = req.media_type
+      if (type && type.match(@json_regex)) ||
+         ["{","["].include?(req.body.read(1))
+        begin
+          req.body.rewind
+          env['rack.json'] = @parser.call(req.body.read)
+        rescue => ex
+          raise BadRequest, "json parse error: #{ex.message}"
```
Comment (Hudon) [inline lib/jersey/middleware/auto_json.rb]: If the content type is specified, the error makes sense, but if we're guessing through `[` and `{`, a parse error should just be treated like our guess was wrong

**Comment by csquared** [comment]
@Hudon @jimpo @aalness rescue nil ftw

**Pull request merged**